### DISCUSSION TITLE
'str' object has no attribute 'text' for `sentencizer` of `xx_ent_wiki_sm`

### DISCUSSION THREAD

**Discussion by ZenBel** [explosion/spaCy#11539] [Help: Other Questions] [answered]
When I run this piece of code:

```
spacy.cli.download("xx_ent_wiki_sm")
nlp = spacy.load("xx_ent_wiki_sm",disable=["tagger", "lemmatizer", "ner", "textcat", "parser"])
nlp.add_pipe("sentencizer")
nlp("hello. goodbye")
```
everything works perfectly. However, when I try to run this piece of code

```
spacy.cli.download("xx_ent_wiki_sm")
nlp = spacy.load("xx_ent_wiki_sm",disable=["tagger", "lemmatizer", "ner", "textcat", "parser"]).add_pipe("sentencizer")
nlp("hello. goodbye")
```

I get the following error: `AttributeError: 'str' object has no attribute 'text'`.

But I don't understand why. Can someone shed some light on it?

Thanks in advance,

Z

**Comment by rmitsch** [comment, marked as answer]
Hi @ZenBel, `Language.add_pipe()` returns a `Callable` (like a `Pipe`), not the `Language` instance itself. See also the [`Language` docs](https://spacy.io/api/language#add_pipe).
So what you are doing here is assigning the `sentencizer` pipe you just created to your `nlp` variable and then calling it.

**Reply by ZenBel** [reply]
I see, thank you for the clarification!
